src/effects/effectchain.cpp

```diff
@@ -151,37 +155,63 @@ void EffectChain::setEnabled(bool enabled) {
 }
 
 void EffectChain::enableForChannel(const ChannelHandleAndGroup& handle_group) {
-    if (!m_bAddedToEngine) {
+    // TODO(Be): remove m_enabledChannels from this class and move this logic
+    // to EffectChainSlot
+    bool bWasAlreadyEnabled = m_enabledChannels.contains(handle_group);
+    if (!bWasAlreadyEnabled) {
+        m_enabledChannels.insert(handle_group);
+    }
+
+    // The allocation of EffectStates below may be expensive, so avoid it if
+    // not needed.
+    if (!m_bAddedToEngine || bWasAlreadyEnabled) {
```

Comment:
Can bWasEnabled ever be true? I am afraid this will lead to inconsistence states, because no one keeps track about the new and old enabled channels. I think this was originally done by the getOrCreate function.

Reply:
What do you mean by inconsistent states? Inconsistent between EffectChainSlot and EffectChain, or inconsistent between the main thread and the engine thread?

The former issue reveals a major design flaw in the main thread side of the effects system. We have lots of state duplicated between Effect/EffectSlot and EffectChain/EffectChainSlot. This is really confusing and hard to maintain and the root reason why the saving/loading XML and initialization processes are so convoluted and brittle. It caused me a lot of trouble working on this. There is no clear separation of responsibilities between the Slot and non-Slot classes. Either Effect & EffectChain should not maintain state aside from m_bAddedToEngine, or Effect & EffectSlot as well as EffectChain & EffectChainSlot should be consolidated into single classes. I am leaning towards the latter. This should be taken care of before implementing support for saving and loading chains, but it is beyond the scope of this PR.

Reply:
If you mean the latter, we need to be able to have inconsistent states between the main thread and engine with the current architecture that supports adding and removing chains from the engine. The main thread is responsible for maintaining the list of enabled input channels. This needs to be propagated to the engine thread when a chain is already loaded to the engine and the correct input channels need to be enabled when loading a new chain. This is overcomplicated and error-prone. I think the capability of adding and removing chains from the engine should be removed. Effect units should be hard coded in the engine. Saving and loading chains should be implemented as saving and loading states, not adding and removing chain objects from the engine. But again, this is beyond the scope of this PR.

Reply:
This way, we loose the existing namespace, and we have to reserve unit indexes for the quick effects and EQs. 
So IMHO there is no value for a change, just extra work and anoying legacy aliases.

Reply:
Your comment doesn't make sense in this context. Did you mean to respond to a different comment?

Reply:
Yes, sorry. Here we need to move the enabled channels to the slot as you suggest in your comments.

Reply:
Okay, I can do that, but it does not really solve the larger architectural issues with the main thread side of the effects system.